pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -188,6 +298,96 @@ def _primitive_bind_call(cls, mps, wires, id=None):
             return type.__call__(cls, mps=mps, wires=wires, id=id)  # pragma: no cover
         return cls._primitive.bind(*mps, wires=wires, id=id)
 
+    def decomposition(self):  # pylint: disable=arguments-differ
+        filtered_hyperparameters = {
+            key: value for key, value in self.hyperparameters.items() if key != "input_wires"
+        }
+        return self.compute_decomposition(
+            self.parameters, wires=self.hyperparameters["input_wires"], **filtered_hyperparameters
+        )
+
+    @staticmethod
+    def compute_decomposition(
+        mps, wires, work_wires, right_canonicalize=False
+    ):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+        The decomposition follows Eq. (23) in `[arXiv:2310.18410] <https://arxiv.org/pdf/2310.18410>`_.
+
+        Args:
+            mps (list[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
+                product of site matrices.
+
+            wires (Sequence[int]): wires that the template acts on
+            work_wires (Sequence[int]): list of extra qubits needed in the decomposition. The maximum permissible bond
+                dimension of the provided MPS is defined as ``2^len(work_wires)``. Default is ``None``.
+            right_canonicalize (bool): Indicates whether a conversion to right-canonical form should be performed
+                to the mps. Default is ``False``.
+
+        Returns:
+            list[.Operator]: Decomposition of the operator
+        """
+
+        if work_wires is None:
+            raise ValueError("The qml.MPSPrep decomposition requires `work_wires` to be specified.")
+
+        bond_dimensions = []
+
+        for i in range(len(mps) - 1):
+            bond_dim = mps[i].shape[-1]
+            bond_dimensions.append(bond_dim)
+
+        max_bond_dimension = max(bond_dimensions)
```

Comment:
If you aren't using `bond_dimensions` for anything else other than finding the max bond dimensions, I recommend implementing max() manually on `bond_dim` for performance.